transactions/views.py

```diff
@@ -469,3 +482,66 @@ def get_queryset(self):
                 Q(transaction__buyer__id=user.id)
                 | Q(account__owner__id=user.id))
         return query
+
+
+class SavedCardsListView(generics.GenericAPIView):
+    """get all a  cards for authenticated user"""
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CardInfoSerializer
+    def get_queryset(self):
+
+        user = self.request.user
+        return CardInfo.active_objects.all_objects().filter(user_id=user.id)
+
+    def get(self, request):
+        serializer = self.serializer_class(self.get_queryset(), many=True)
+        if not serializer.data:
+            response = {
+                "data": {
+                    "message": "You don't have any card saved"
+                }
+            }
+            status_code = status.HTTP_404_NOT_FOUND
+        else:
+            response = {
+                "data": {
+                    "saved_cards": serializer.data,
+                    "message": "Cards retrieved",
+                }
+            }
+            status_code = status.HTTP_200_OK
+        return Response(response, status=status_code)
+
+
+class DeleteSavedCardView(generics.GenericAPIView):
+    """ Remove existing card """
+
+    permission_classes = (IsAuthenticated, IsCardOwner)
+
+    def get_object(self, request, card_info_id):
+        """
+        Handles getting a specific card
+
+        :param request:
+        :param user_id:
```

Comment:
**user_id** is not one of the parameters. Also, **card_info_id** is not captured in the docstring. In addition, the **get_method** we are overriding expects one parameter, **self**.